Import classes and modules in __init__.py of each package

gonzaiva opened the pull request:
Classes and modules defined in the TensorFlow tree were imported in the file `__init__.py` of each package to make the current package-level imports shorter.

Commit:
Import classes and modules in __init__.py of each package (#36)

Classes and modules defined in the TensorFlow tree were imported
in the file __init__.py of each package to make the current
package-level imports shorter.

Commit:
Format all files modified with PEP8 using YAPF

This changes are required to avoid conflicts with Travis CI
in GitHub.

Commit:
Remove changes pushed by accident

It was double checked that only one external test file and enabling
plot in trpo_cartpole were the changes pushed by accident.

ryanjulian commented on sandbox/rocky/tf/algos/__init__.py:
```diff
@@ -1 +1,4 @@
-
+from .batch_polopt import BatchPolopt
```
https://www.python.org/dev/peps/pep-0008/#id23

Please avoid relative imports

ryanjulian commented on examples/trpo_gym_tf_cartpole.py:
```diff
@@ -34,9 +33,4 @@
-    snapshot_mode="last",
-    seed=1
-)
+run_experiment_lite(algo.train(), n_parallel=1, snapshot_mode="last", seed=1)
```
This is ugly but probably what was demanded by YAPF. It's definitely better to have the list style.

You can disable YAPF when it is wrong using this comment.
```python
run_experiment_lite(
    algo.train(),
    n_parallel=1,
    snapshot_mode="last",
    seed=1
) # yapf: disable
```

ryanjulian commented on sandbox/rocky/tf/core/__init__.py:
```diff
@@ -1 +1,7 @@
-
+from .parameterized import Parameterized
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/distributions/__init__.py:
```diff
@@ -1 +1,6 @@
-
+from .bernoulli import Bernoulli
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/envs/__init__.py:
```diff
@@ -1 +1,3 @@
-
+from .base import TfEnv
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/launchers/vpg_cartpole.py:
```diff
@@ -25,10 +24,5 @@
-        )
-    )
-)
+    optimizer_args=dict(tf_optimizer_args=dict(learning_rate=0.01, )))
```
# yapf: disable

ryanjulian commented on sandbox/rocky/tf/optimizers/__init__.py:
```diff
@@ -1 +1,5 @@
-
+from .conjugate_gradient_optimizer import ConjugateGradientOptimizer
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/policies/__init__.py:
```diff
@@ -1 +1,6 @@
-
+from .base import Policy
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/q_functions/__init__.py:
```diff
@@ -0,0 +1 @@
+from .base import QFunction
```
no relative imports

ryanjulian commented on sandbox/rocky/tf/samplers/__init__.py:
```diff
@@ -1 +1,2 @@
-
+from .batch_sampler import BatchSampler
```
no relative imports

ryanjulian reviewed (changes requested):
Great work, this definitely makes the whole codebase more readable with one small change.

Some comments:
1. __Remove the YAPF formatting__
I apologize for not anticipating this earlier. The intention of the policy of only reformatting edited files is to gradually reformat the codebase while we maintain it, where maintain is defined as "touching a file." A large-scale change like this, which touches every file in a small way, violates the spirit of the rule. It implicitly makes this a (reformat everything + change imports) PR, but we only want the PR to do one thing. __So for this change, please only change the import lines and don't bother with YAPF or PEP8 for the files.__ In the future we might just push one giant PR which YAPFifys the whole codebase. We will ignore the CI and push the change through. Again, I apologize for the inconvenience.

1. __Do it for the whole codebase, not just the TF tree__
It's important to be consistent. We are maintaining all of rllab, even though we focus on TF mostly.

1. __No relative imports__
PEP8 recommends not using relative imports except in special circumstances. The logic behind this is that relative imports obscure the absolute package path when a non-developer is browsing the repository, and they also make bugs from refactors and renamings harder to find. I don't think this circumstance is special enough. https://www.python.org/dev/peps/pep-0008/#id23

Commit:
Revert "Remove changes pushed by accident"

This reverts commit 3256edde11a7a004f67a17be424b5c05fac6e995.

Commit:
Revert "Format all files modified with PEP8 using YAPF"

This reverts commit de0dfa9f27979bac10b1d42c04e3c30df8306ff2.

Commit:
Change relative imports in __init__ files for absolute imports

Relative imports can bring issues when the package hierarchy and
module is browsed by non-developers, and other problems with
re-factoring and renaming.

ghost commented:
@ryanjulian I reverted the changes concerning PEP8 formatting and changed the relative imports for absolute imports. Also I had to merge one commit from the integration branch.
Please review the changes again.

ryanjulian reviewed (dismissed):
LGTM

ryanjulian commented:
I was just about to merge this, but realized it's only for the TF tree. Can you do it for *all* of rllab?

CatherineSue committed:
Merge with origin/integration

List of merged commits:
bb2475f65ddae62c020c647f87634c5971014c1d

Commit:
Convenience imports for Theano branch

Classes and modules defined in the Theano tree were imported in the file
__init__.py of each package to make the current package-level imports
shorter.

Commit:
Merge branch 'integration' into convenience_imports

ghost commented:
@ryanjulian the change requested is done, please review.
Again, the CI build is failing due to check_yapf.sh:
![image](https://user-images.githubusercontent.com/35857569/40397555-329cd08a-5de8-11e8-8d9a-8314b1a25701.png)

ryanjulian commented:
The latest run seems to be executing fine.

https://travis-ci.org/ryanjulian/rllab/builds/382443801?utm_source=github_status&utm_medium=notification

ryanjulian commented:
Fixes #66
